### PR TITLE
[FIX] website: Handle matching words with point or slash

### PR DESCRIPTION
**Pull request by mohamedlaminelalmi**
[FIX] website: Handle matching words with point or slash
    
Fix an issue where products with internal references containing point
or slash were not recognized as direct matches in the website searchbar.
Instead, fuzzy search was used to find similar words.

This commit addresses the problem by fixing the regular expression
used for matching words in Odoo's search functionality.

### CODE REVIEW

**Review by mehjabinfarsana — COMMENTED** [review]
hi @mohamedlaminelalmi ,
in case if you haven't signed the CLA yet, as i see the runbot test is getting failed, please sign the cla:  https://github.com/odoo/odoo/blob/16.0/doc/cla/sign-cla.md

thanks

**Comment by mohamedlaminelalmi** [comment]
@mehjabinfarsana Thank you. The CLA is signed 🥇

**Comment by rdeodoo** [comment]
@bso-odoo Hello Benoit, maybe you can/want to have a look here?

@robodoo delegate=@bso-odoo

File: addons/website/models/website.py [hunk 1]
```diff
@@ -1661,7 +1661,7 @@ def _trigram_enumerate_words(self, search_details, search, limit):
         :param limit: maximum number of records fetched per model to build the word list
         :return: yields words
         """
-        match_pattern = r'[\w-]{%s,}' % min(4, len(search) - 3)
+        match_pattern = r'[\w./-]{%s,}' % min(4, len(search) - 3)
```
Comment (bso-odoo) [inline addons/website/models/website.py:1664]: The same pattern update should be applied within `_basic_enumerate_words` too for consistency.

**Review by bso-odoo — COMMENTED** [review]
Hi, thank you for your commit ! Here is my review.
I just validated with the product owner and he also agrees your suggestion makes sense.

The commit message mentions underscore, but it was already handled by `\w`. :wink: 
Also, try to limit your commit message to 72 columns (e.g. just say `... with point or slash` to make it shorter).

I believe the test that was introduced with <https://github.com/odoo/odoo/commit/e9ea0692bb39adbc471b656fb47d08e6745d66b4> is sufficient, but feel free to add a test with your specific use case if you want.

File: addons/website/models/website.py (hunk 1 above)
Comment (mohamedlaminelalmi) [reply addons/website/models/website.py:1664]: @bso-odoo The updates are also applied on _basic_enumerate_words  ;) 
Thank you !

**Comment by mohamedlaminelalmi** [comment]
@bso-odoo The commit message has been updated.

Thanks again !

**Comment by bso-odoo** [comment]
> @bso-odoo The commit message has been updated.

Thanks for the update !
The message now says `with with` :wink:.
I'll let you fix it before merging.

**Comment by mohamedlaminelalmi** [comment]
@bst-odoo the "with with" is fixed 😉
Thank you!

**Comment by bso-odoo** [comment]
> @bso-odoo the "with with" is fixed wink Thank you!

I still see it inside the c8806739fbf1a2e17ba5ab2100b1e9ad2b0d86c0 commit. It does not seem like you pushed your last change.

**Comment by mohamedlaminelalmi** [comment]
@bso-odoo Sorry for that 😶‍🌫️ 
It should be OK now 😉

**Comment by bso-odoo** [comment]
Thanks a lot for your contribution ! :clap:  
Sorry for the iterations :roll_eyes:

**Pull request closed without merging**